go/private/library.bzl

```diff
@@ -209,38 +193,31 @@ def go_importpath(ctx):
     path = path[1:]
   return path
 
-def get_gc_goopts(ctx):
-  gc_goopts = tuple(ctx.attr.gc_goopts)
-  if ctx.attr.library:
-    gc_goopts += ctx.attr.library[GoLibrary].gc_goopts
-  return gc_goopts
-
-def emit_go_compile_action(ctx, sources, libs, lib_paths, direct_paths, out_object, gc_goopts):
+def emit_go_compile_action(ctx, sources, golibs, mode, out_object, gc_goopts):
```

Comment:
Same comments about `mode`, doc string, and abstraction as `emit_go_link_action`.